Tools/autotest/arducopter.py

```diff
@@ -965,6 +965,30 @@ def setup_rc(mavproxy):
     mavproxy.send('rc 3 1000\n')
 
 
+def fly_throw_mode(mavproxy, mav):
+        # test boomerang mode:
+        progress("Throwing vehicle away")
+        set_parameter(mavproxy, "THROW_NEXTMODE", 6)
+        set_parameter(mavproxy, "SIM_SHOVE_Z", -30)
+        set_parameter(mavproxy, "SIM_SHOVE_X", -20)
+        mavproxy.send('mode throw\n')
+        wait_mode(mav, 'THROW')
+        mavproxy.send('arm throttle\n')
+        mav.motors_armed_wait()
+        set_parameter(mavproxy, "SIM_SHOVE_TIME", 500)
+        tstart = get_sim_time(mav)
+        wait_mode(mav, 'RTL')
+        max_good_tdelta = 5
```

Comment:
Apparently this isn't good enough, it took 7 🙂 In that sense, I think it would be preferable to check if it didn't crash rather than how much time it took for RTL to engage.

Reply:
The trigger for writing this test is that I took my real vehicle out and threw it in the air and it didn't come down.  It stayed in throw mode, hovering rather nicely.

Well, it did - by turning off all the motors, seemingly when I changed modes.  Thus my desire for having a timeout here.

Reply:
Alright, what about checking for a crash and having a bigger timeout to check it changed to the desired mode?